fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/StarRating.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.round
+import kotlin.math.roundToInt
+
+private const val NUM_STARS = 5
+private const val MAX_RATING = 5f
+private const val MIN_RATING = 0f
+
+/**
+ * UI for displaying star rating bar with maximum 5 stars.
+ */
+@Composable
+fun StarRating(
+    modifier: Modifier = Modifier,
+    value: Float,
+) {
+    val rating: Float = remember(value) {
+        max(min(MAX_RATING, value), MIN_RATING).roundToNearestHalf()
+    }
+    val contentDescription = contentDescription(rating = rating)
+
+    Row(
+        modifier = modifier.semantics {
+            this.contentDescription = contentDescription
+        },
+        horizontalArrangement = Arrangement.spacedBy(4.dp),
+    ) {
+        repeat(NUM_STARS) {
+            val star = if (it < rating && it + 1 > rating) {
+                R.drawable.ic_star_half
+            } else if (it < rating) {
+                R.drawable.ic_bookmark_filled
+            } else {
+                R.drawable.ic_bookmark_outline
+            }
+
+            Image(
+                painter = painterResource(id = star),
+                contentDescription = null,
+            )
```

Comment:
do we need to tint the stars?

Reply:
Good question, yes, It comes from the vectors itself

Reply:
Gotcha gotcha. I know it's a little pedantic, since the vector is getting `textPrimary`, which is technically the same as `iconPrimary`, but to cover ourselves in case the design system ever changes, can we use `FirefoxTheme.colors.iconPrimary` here instead?

Reply:
no it's reasonable, updated 👍